confmap/expand.go

```diff
@@ -138,11 +142,26 @@ func toString(input any) (string, error) {
 	}
 }
 
-func (mr *Resolver) expandURI(ctx context.Context, uri string) (any, bool, error) {
-	lURI, err := newLocation(uri[2 : len(uri)-1])
-	if err != nil {
-		return nil, false, err
+func (mr *Resolver) expandURI(ctx context.Context, input string) (any, bool, error) {
+	// strip ${ and }
+	uri := input[2 : len(input)-1]
+
+	var lURI location
+	var err error
+	// If no ':' then we have no scheme, check if there is a default scheme configured
+	if !strings.Contains(uri, ":") {
+		// If we have no default scheme there is nothing that can be expanded, return the original input unchanged
+		if mr.defaultScheme == "" {
+			return input, false, nil
+		}
+		lURI = location{scheme: mr.defaultScheme, opaqueValue: uri}
```

Comment:
I would rather make `uri = mr.defaultScheme + ":" + uri` then apply same logic of using newLocation. Ensures we have consistent checks.